fabbwled-frontend/src/components/Inventory.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import {
+  Box,
+  Button,
+  IconButton,
+  Input,
+  List,
+  ListItemButton,
+  ListItemText,
+  Modal,
+  Typography,
+} from "@mui/material";
+import { Close } from "@mui/icons-material";
+import background from "../../public/inventory-background.png";
+
+interface Item {
+  id: number;
+  name: string;
+}
+
+type InventoryProps = {
+  onAction: () => void;
+};
+
+const Inventory = ({ onAction }: InventoryProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [selectedItem, setSelectedItem] = useState<Item | null>(null);
+
+  const inventory: Item[] = [
+    { id: 1, name: "Item 1" },
+    { id: 2, name: "Item 2" },
+    { id: 3, name: "Item 3" },
+    { id: 4, name: "Item 4" },
+    { id: 5, name: "Item 5" },
+    { id: 6, name: "Item 6" },
+    { id: 7, name: "Item 7" },
+  ];
+
+  const filteredItems = inventory.filter((item) =>
+    item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+  );
+
+  const handleItemClick = (item: Item) => {
+    setSelectedItem(item);
+  };
+
+  const handleActionClick = () => {
+    if (selectedItem) {
```

Comment:
change this to a guard statement
```suggestion
    if (!selectedItem) {
       return;
    }
```